### PR TITLE
[ci][core] Run microbenchmark for python 3.11

### PR DESCRIPTION
**Pull request by rickyyx**
## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by can-anyscale** [comment]
Probably need to update here too https://github.com/ray-project/ray/blob/master/release/ray_release/byod/build_ray.py#L15 to run tests on PR

**Review by aslonnie — COMMENTED** [review]
lgtm; will leave this for @can-anyscale to approve.

**Comment by can-anyscale** [comment]
Look like release test is still failing

**Comment by rickyyx** [comment]
Test passes: https://buildkite.com/ray-project/release-tests-pr/builds/49200#0189fc99-8724-4235-9ed8-f5a38c013a64

File: release/release_tests.yaml [hunk 1]
```diff
@@ -4903,6 +4921,21 @@
       cluster:
         cluster_compute: tpl_64_gce.yaml
 
+- name: microbenchmark_311
```
Comment (can-anyscale) [inline release/release_tests.yaml:4924]: instead of creating a new blob, can we use variations in the blob right above?

**Review by can-anyscale — APPROVED** [review]

File: release/release_tests.yaml (hunk 1 above)
Comment (rickyyx) [reply release/release_tests.yaml]: good idea. let me try

File: release/release_tests.yaml [hunk 2]
```diff
@@ -2733,6 +2745,8 @@
 
   cluster:
     byod:
+      pip:
```
Comment (rickyyx) [inline release/release_tests.yaml:2748]: are these changes needed to run python 3.11 actually?

File: release/release_tests.yaml [hunk 3]
```diff
@@ -4902,6 +4920,8 @@
       frequency: manual
       cluster:
         cluster_compute: tpl_64_gce.yaml
+    - __suffix__: aws.py311
```
Comment (rickyyx) [inline release/release_tests.yaml:4923]: added a variance.

File: release/release_tests.yaml (hunk 2 above)
Comment (can-anyscale) [reply release/release_tests.yaml:2748]: this autorom is not compatible with 3.11, so I need to pull it out of the common Docker image

Comment (rickyyx) [reply release/release_tests.yaml:2748]: should i add a TODO to track it? do we plan to add it back later?

Comment (can-anyscale) [reply release/release_tests.yaml:2748]: a TODO sounds good, thankks! we can add them back once autorom supports 3.11
